guix: Clean up manifest

hebasto opened the pull request:
This PR cleans up the `contrib/guix/manifest.scm` in the following way:
- Unneeded for a successful build modules have be dropped.
- Some modules have been enhanced with `#:select` clauses, which improves maintainability (see the commit message for details).

hebasto committed:
guix: Unify fetch methods

hebasto committed:
guix: Drop unneeded modules

TheCharlatan commented on contrib/guix/manifest.scm:
```diff
@@ -1,27 +1,27 @@
              (gnu packages gcc)
-             (gnu packages installers)
+             ((gnu packages installers) #:select (nsis-x86_64))
              (gnu packages linux)
```
Would it make sense to add `((gnu packages linux) #:select (util-linux linux-libre-headers-5.15))`?

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/27811#issuecomment-1578197211).

Sjors commented on contrib/guix/manifest.scm:
```diff
@@ -257,9 +256,6 @@ and abstract ELF, PE and MachO formats.")
      `(("openssl", openssl)))
-    (arguments
-     '(#:configure-flags
-        (list "-DCMAKE_DISABLE_FIND_PACKAGE_CURL=TRUE")))
```
641897a83dc9d40b618efbae67c3beb90a1f34f8: What did this do?

hebasto replied:
Please see https://github.com/bitcoin/bitcoin/pull/27779.

hebasto replied:
> What did this do?

Effectively, [nothing](https://github.com/bitcoin/bitcoin/pull/27779#issuecomment-1572754794).

hebasto committed:
guix: Specify symbols in modules explicitly

This change improves the maintainability of the manifest:
(1) It allows to remove the module when the specified symbols are no
    longer used.
(2) It prevents accidental use of other symbols, such as `bash`
    instead of `bash-minimal`.

hebasto commented:
Updated 86739705406edbe89bfb039a63fb33361990d673 -> 404a0e98dd964af4db338843f65ca3a63429db17 ([pr27811.02](https://github.com/hebasto/bitcoin/commits/pr27811.02) -> [pr27811.03](https://github.com/hebasto/bitcoin/commits/pr27811.03), [diff](https://github.com/hebasto/bitcoin/compare/pr27811.02..pr27811.03)):

- addressed @TheCharlatan's [comment](https://github.com/bitcoin/bitcoin/pull/27811#discussion_r1219147914)

hebasto commented:
Rebased 404a0e98dd964af4db338843f65ca3a63429db17 -> 77414d50d6f71bfa1e7c04ec672167cf0cf0dedc ([pr27811.03](https://github.com/hebasto/bitcoin/commits/pr27811.03) -> [pr27811.04](https://github.com/hebasto/bitcoin/commits/pr27811.04)) on top of the just merged https://github.com/bitcoin/bitcoin/pull/27779.

TheCharlatan commented:
Guix build:
```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
c4f2c29c7b5e1ab211ea38358517cf6c2a1e154e5d2c0d036e7750a8f5f5555a  guix-build-77414d50d6f7/output/aarch64-linux-gnu/SHA256SUMS.part
c902944cce64fe7f2cee1242f18b35dbe1c5ce515e921c08c9bfebb34d0c28ca  guix-build-77414d50d6f7/output/aarch64-linux-gnu/bitcoin-77414d50d6f7-aarch64-linux-gnu-debug.tar.gz
3aad68ed102f2f1a18217b3d2b2b32ba51f266c03b971f05f8bebd160bbf7d75  guix-build-77414d50d6f7/output/aarch64-linux-gnu/bitcoin-77414d50d6f7-aarch64-linux-gnu.tar.gz
36737484c80cdfc108c3800f34c7a56aaa4d3c53339de1e876ea4235270b1bfd  guix-build-77414d50d6f7/output/arm-linux-gnueabihf/SHA256SUMS.part
82ad6d7382be867fcdadbc0803dbd05824da6a54112024d1830f3e4b22b00a0c  guix-build-77414d50d6f7/output/arm-linux-gnueabihf/bitcoin-77414d50d6f7-arm-linux-gnueabihf-debug.tar.gz
44482ef8f9819f4f35d7dd23a9fed563f14c85afde70454e50fb3d56915f1262  guix-build-77414d50d6f7/output/arm-linux-gnueabihf/bitcoin-77414d50d6f7-arm-linux-gnueabihf.tar.gz
b57c6e02c66b650add22547493fb28ac105a0cb8de675bd1a944ceba1197523b  guix-build-77414d50d6f7/output/arm64-apple-darwin/SHA256SUMS.part
d8fc34cb0192d600fae15fd8025c73f48acd19271bba998cd05c3559ffa6d521  guix-build-77414d50d6f7/output/arm64-apple-darwin/bitcoin-77414d50d6f7-arm64-apple-darwin-unsigned.dmg
23339a33c1586e562ec1998b42d84fd2b358e2318abdfb10f4bb15f356066cf0  guix-build-77414d50d6f7/output/arm64-apple-darwin/bitcoin-77414d50d6f7-arm64-apple-darwin-unsigned.tar.gz
4a94b2d7d562e80f12479103eb0a934d0439a4535e3c6f7fa7fca04a9f4095ec  guix-build-77414d50d6f7/output/arm64-apple-darwin/bitcoin-77414d50d6f7-arm64-apple-darwin.tar.gz
afa6d27b36dcd73b6f03b8018190f1522f4ad98819ad672335226d6fedc7ebc0  guix-build-77414d50d6f7/output/dist-archive/bitcoin-77414d50d6f7.tar.gz
5d66c44562a6d9ad643def223180bf710921f92b3d64ba2c75ba4a3d66a6a0b3  guix-build-77414d50d6f7/output/powerpc64-linux-gnu/SHA256SUMS.part
44606d6b4684a69cbe1a3b3231f015d265b1339847291f33df4f81d4c6b60724  guix-build-77414d50d6f7/output/powerpc64-linux-gnu/bitcoin-77414d50d6f7-powerpc64-linux-gnu-debug.tar.gz
8e01b6920cd5d7ac96bfc0ca58568eb05c34d10cb663d75d0908050491bcbd89  guix-build-77414d50d6f7/output/powerpc64-linux-gnu/bitcoin-77414d50d6f7-powerpc64-linux-gnu.tar.gz
096509b1297a4f1db2d9c3af32cbd25748a6e9a2b0560f02546a829121000c44  guix-build-77414d50d6f7/output/powerpc64le-linux-gnu/SHA256SUMS.part
f0614f5743d765bd8f41a05880283ede5ba63f498681434cbbb75c1b4b88ebb2  guix-build-77414d50d6f7/output/powerpc64le-linux-gnu/bitcoin-77414d50d6f7-powerpc64le-linux-gnu-debug.tar.gz
b27f979757361597d0534f7fbb4d6f0f45ec4faf3c40e29a99610ae02da5acfa  guix-build-77414d50d6f7/output/powerpc64le-linux-gnu/bitcoin-77414d50d6f7-powerpc64le-linux-gnu.tar.gz
e5c292afacf03043c5c78a4ffa610bf264675d6343a9e80f633e095d1d74eaac  guix-build-77414d50d6f7/output/riscv64-linux-gnu/SHA256SUMS.part
5be6e791602bb9e5a7b60262394154eaa6ba4a65853bdebd316677e1e91159fe  guix-build-77414d50d6f7/output/riscv64-linux-gnu/bitcoin-77414d50d6f7-riscv64-linux-gnu-debug.tar.gz
f7d76abcb8aac0f3ad9f048f5ae08322244f02ca19b2826bfee73e534b4b538f  guix-build-77414d50d6f7/output/riscv64-linux-gnu/bitcoin-77414d50d6f7-riscv64-linux-gnu.tar.gz
1c639178ac99a00b90527bd8e4813f1ca3f6be13eddfd299f0623ed8781e3508  guix-build-77414d50d6f7/output/x86_64-apple-darwin/SHA256SUMS.part
a5a882211cc670b57265bcc215a3ba7cbce346c3d286291b6f9ff9fc160a975c  guix-build-77414d50d6f7/output/x86_64-apple-darwin/bitcoin-77414d50d6f7-x86_64-apple-darwin-unsigned.dmg
5336bbe4427dbcabbba83abc20a139e7ad38a0f4948b9f8e6462e6dd5054fb99  guix-build-77414d50d6f7/output/x86_64-apple-darwin/bitcoin-77414d50d6f7-x86_64-apple-darwin-unsigned.tar.gz
521c961de0be73987ae26e4806bd758496c3f1235f7563714a8cf46c10dd6eb0  guix-build-77414d50d6f7/output/x86_64-apple-darwin/bitcoin-77414d50d6f7-x86_64-apple-darwin.tar.gz
fda9e9f2bd981e5b8592dfd2beb4a055a3c8355a78977d7f72d47ba912471e0a  guix-build-77414d50d6f7/output/x86_64-linux-gnu/SHA256SUMS.part
aaa4b1dcec736d192841b8d0a0ccec4177053f0a42ec14233585dca896d6d22d  guix-build-77414d50d6f7/output/x86_64-linux-gnu/bitcoin-77414d50d6f7-x86_64-linux-gnu-debug.tar.gz
00ef9ad60dda7308e8b0c6e6133f11202f552c522ac531c87a6e8d8f8e59d23f  guix-build-77414d50d6f7/output/x86_64-linux-gnu/bitcoin-77414d50d6f7-x86_64-linux-gnu.tar.gz
dfa91bb10a79338f9e9ed8c1ac033d555fd205ba5c74c897d4e17857a30270f4  guix-build-77414d50d6f7/output/x86_64-w64-mingw32/SHA256SUMS.part
27c191ce58b1f851f8ab5899614d5297b39cea8a5f3fb7d6c5c53309721f2610  guix-build-77414d50d6f7/output/x86_64-w64-mingw32/bitcoin-77414d50d6f7-win64-debug.zip
e9a1663d47b2ad895952f4b12159980f0175ae6aefb4318766daeabfb738ff44  guix-build-77414d50d6f7/output/x86_64-w64-mingw32/bitcoin-77414d50d6f7-win64-setup-unsigned.exe
0c98d10223054262749bcf7480e41986bd43a5f2ce5d3b31ea530678d72e26e2  guix-build-77414d50d6f7/output/x86_64-w64-mingw32/bitcoin-77414d50d6f7-win64-unsigned.tar.gz
4fbf17fc4f62c8d2cfecda356d4b99f67a36eb6a136c784b2308d657fed1c8fb  guix-build-77414d50d6f7/output/x86_64-w64-mingw32/bitcoin-77414d50d6f7-win64.zip
```

Sjors commented:
My guix build matches @TheCharlatan

hebasto commented:
I'll be able to provide my own Guix build hashes tomorrow only.

hebasto commented:
My Guix build:
```
c4f2c29c7b5e1ab211ea38358517cf6c2a1e154e5d2c0d036e7750a8f5f5555a  guix-build-77414d50d6f7/output/aarch64-linux-gnu/SHA256SUMS.part
c902944cce64fe7f2cee1242f18b35dbe1c5ce515e921c08c9bfebb34d0c28ca  guix-build-77414d50d6f7/output/aarch64-linux-gnu/bitcoin-77414d50d6f7-aarch64-linux-gnu-debug.tar.gz
3aad68ed102f2f1a18217b3d2b2b32ba51f266c03b971f05f8bebd160bbf7d75  guix-build-77414d50d6f7/output/aarch64-linux-gnu/bitcoin-77414d50d6f7-aarch64-linux-gnu.tar.gz
36737484c80cdfc108c3800f34c7a56aaa4d3c53339de1e876ea4235270b1bfd  guix-build-77414d50d6f7/output/arm-linux-gnueabihf/SHA256SUMS.part
82ad6d7382be867fcdadbc0803dbd05824da6a54112024d1830f3e4b22b00a0c  guix-build-77414d50d6f7/output/arm-linux-gnueabihf/bitcoin-77414d50d6f7-arm-linux-gnueabihf-debug.tar.gz
44482ef8f9819f4f35d7dd23a9fed563f14c85afde70454e50fb3d56915f1262  guix-build-77414d50d6f7/output/arm-linux-gnueabihf/bitcoin-77414d50d6f7-arm-linux-gnueabihf.tar.gz
b57c6e02c66b650add22547493fb28ac105a0cb8de675bd1a944ceba1197523b  guix-build-77414d50d6f7/output/arm64-apple-darwin/SHA256SUMS.part
d8fc34cb0192d600fae15fd8025c73f48acd19271bba998cd05c3559ffa6d521  guix-build-77414d50d6f7/output/arm64-apple-darwin/bitcoin-77414d50d6f7-arm64-apple-darwin-unsigned.dmg
23339a33c1586e562ec1998b42d84fd2b358e2318abdfb10f4bb15f356066cf0  guix-build-77414d50d6f7/output/arm64-apple-darwin/bitcoin-77414d50d6f7-arm64-apple-darwin-unsigned.tar.gz
4a94b2d7d562e80f12479103eb0a934d0439a4535e3c6f7fa7fca04a9f4095ec  guix-build-77414d50d6f7/output/arm64-apple-darwin/bitcoin-77414d50d6f7-arm64-apple-darwin.tar.gz
afa6d27b36dcd73b6f03b8018190f1522f4ad98819ad672335226d6fedc7ebc0  guix-build-77414d50d6f7/output/dist-archive/bitcoin-77414d50d6f7.tar.gz
5d66c44562a6d9ad643def223180bf710921f92b3d64ba2c75ba4a3d66a6a0b3  guix-build-77414d50d6f7/output/powerpc64-linux-gnu/SHA256SUMS.part
44606d6b4684a69cbe1a3b3231f015d265b1339847291f33df4f81d4c6b60724  guix-build-77414d50d6f7/output/powerpc64-linux-gnu/bitcoin-77414d50d6f7-powerpc64-linux-gnu-debug.tar.gz
8e01b6920cd5d7ac96bfc0ca58568eb05c34d10cb663d75d0908050491bcbd89  guix-build-77414d50d6f7/output/powerpc64-linux-gnu/bitcoin-77414d50d6f7-powerpc64-linux-gnu.tar.gz
096509b1297a4f1db2d9c3af32cbd25748a6e9a2b0560f02546a829121000c44  guix-build-77414d50d6f7/output/powerpc64le-linux-gnu/SHA256SUMS.part
f0614f5743d765bd8f41a05880283ede5ba63f498681434cbbb75c1b4b88ebb2  guix-build-77414d50d6f7/output/powerpc64le-linux-gnu/bitcoin-77414d50d6f7-powerpc64le-linux-gnu-debug.tar.gz
b27f979757361597d0534f7fbb4d6f0f45ec4faf3c40e29a99610ae02da5acfa  guix-build-77414d50d6f7/output/powerpc64le-linux-gnu/bitcoin-77414d50d6f7-powerpc64le-linux-gnu.tar.gz
e5c292afacf03043c5c78a4ffa610bf264675d6343a9e80f633e095d1d74eaac  guix-build-77414d50d6f7/output/riscv64-linux-gnu/SHA256SUMS.part
5be6e791602bb9e5a7b60262394154eaa6ba4a65853bdebd316677e1e91159fe  guix-build-77414d50d6f7/output/riscv64-linux-gnu/bitcoin-77414d50d6f7-riscv64-linux-gnu-debug.tar.gz
f7d76abcb8aac0f3ad9f048f5ae08322244f02ca19b2826bfee73e534b4b538f  guix-build-77414d50d6f7/output/riscv64-linux-gnu/bitcoin-77414d50d6f7-riscv64-linux-gnu.tar.gz
1c639178ac99a00b90527bd8e4813f1ca3f6be13eddfd299f0623ed8781e3508  guix-build-77414d50d6f7/output/x86_64-apple-darwin/SHA256SUMS.part
a5a882211cc670b57265bcc215a3ba7cbce346c3d286291b6f9ff9fc160a975c  guix-build-77414d50d6f7/output/x86_64-apple-darwin/bitcoin-77414d50d6f7-x86_64-apple-darwin-unsigned.dmg
5336bbe4427dbcabbba83abc20a139e7ad38a0f4948b9f8e6462e6dd5054fb99  guix-build-77414d50d6f7/output/x86_64-apple-darwin/bitcoin-77414d50d6f7-x86_64-apple-darwin-unsigned.tar.gz
521c961de0be73987ae26e4806bd758496c3f1235f7563714a8cf46c10dd6eb0  guix-build-77414d50d6f7/output/x86_64-apple-darwin/bitcoin-77414d50d6f7-x86_64-apple-darwin.tar.gz
fda9e9f2bd981e5b8592dfd2beb4a055a3c8355a78977d7f72d47ba912471e0a  guix-build-77414d50d6f7/output/x86_64-linux-gnu/SHA256SUMS.part
aaa4b1dcec736d192841b8d0a0ccec4177053f0a42ec14233585dca896d6d22d  guix-build-77414d50d6f7/output/x86_64-linux-gnu/bitcoin-77414d50d6f7-x86_64-linux-gnu-debug.tar.gz
00ef9ad60dda7308e8b0c6e6133f11202f552c522ac531c87a6e8d8f8e59d23f  guix-build-77414d50d6f7/output/x86_64-linux-gnu/bitcoin-77414d50d6f7-x86_64-linux-gnu.tar.gz
dfa91bb10a79338f9e9ed8c1ac033d555fd205ba5c74c897d4e17857a30270f4  guix-build-77414d50d6f7/output/x86_64-w64-mingw32/SHA256SUMS.part
27c191ce58b1f851f8ab5899614d5297b39cea8a5f3fb7d6c5c53309721f2610  guix-build-77414d50d6f7/output/x86_64-w64-mingw32/bitcoin-77414d50d6f7-win64-debug.zip
e9a1663d47b2ad895952f4b12159980f0175ae6aefb4318766daeabfb738ff44  guix-build-77414d50d6f7/output/x86_64-w64-mingw32/bitcoin-77414d50d6f7-win64-setup-unsigned.exe
0c98d10223054262749bcf7480e41986bd43a5f2ce5d3b31ea530678d72e26e2  guix-build-77414d50d6f7/output/x86_64-w64-mingw32/bitcoin-77414d50d6f7-win64-unsigned.tar.gz
4fbf17fc4f62c8d2cfecda356d4b99f67a36eb6a136c784b2308d657fed1c8fb  guix-build-77414d50d6f7/output/x86_64-w64-mingw32/bitcoin-77414d50d6f7-win64.zip
```

TheCharlatan commented on contrib/guix/manifest.scm:
```diff
@@ -247,12 +246,13 @@ and abstract ELF, PE and MachO formats.")
     (name "osslsigncode")
     (version "2.5")
     (source (origin
-              (method url-fetch)
```
Is there a good reason for using `git-fetch` beyond being more consistent? Seems like this is just less efficient, because we get and hash the entire repository instead of just an archive. Probably all the other cases should be using `url-fetch` instead, or at least with the exception of the sourceware repository?

hebasto replied:
> Probably all the other cases should be using `url-fetch` instead, or at least with the exception of the sourceware repository?

It seems the opposite approach is preferable according to [that](https://lists.gnu.org/archive/html/guix-devel/2020-05/msg00224.html) discussion. Anyway, I'm happy to drop that commit if it is indeed controversial.

Sjors replied:
Can it use a `--depth 1` fetch?

TheCharlatan replied:
No, specifying `depth` is not supported afaict.  I'm ok with moving ahead with this as is. I don't think the reasoning provided in the linked posts really applies here, but would still be good follow best practices.

tobtoht replied:
I was reminded of this discussion after reading about the recent XZ backdoor (context [here](https://www.openwall.com/lists/oss-security/2024/03/29/4) and [here](https://boehs.org/node/everything-i-know-about-the-xz-backdoor), I'm sure you've all seen it by now). 

>One portion of the backdoor is *solely in the distributed tarballs*.

Specifically, a small change in the upstream source of `m4/build-to-host.m4`, which:

> injected an obfuscated script from the files committed [in the source tree] to be “executed at the end of configure”.

Many projects that use the autotools build system include autogenerated scripts in their signed release tarballs (so consumers can simply run `./configure`). This makes it hard to compare the contents of these tarballs with their git sources (where I believe changes enjoy more scrutiny). For example, try diffing expat's tarball (which Bitcoin uses in `depends`) with its git repository checked out to the same version. It contains 100k+ lines of unauditable autogenerated code soup.

Critically, in the case of XZ, running `autoreconf -fi` would not have prevented the backdoor from being inserted.

The discussion on the linked guix-devel list did not specifically mention the supply chain attack angle, so I wanted to bring it up here. Going with `git-fetch` for the manifest definitely seems to have been the right choice. It may also be worth exploring whether depends should be able to fetch from git repositories as well or automatically clean up autogenerated files in the preprocess stage to somewhat mitigate this attack vector.

fanquake commented:
Concept ACK - have tested this as part of some other changes.

hebasto commented:
Rebased due to the conflict with #27813.

TheCharlatan commented:
ACK a51d7abf1e13c532c7acf437c3a65a9511b44987

Guix builds:
```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
6edd213a90a043d09da144b99332ab16a320774098d35de28464262b25f260b2  guix-build-a51d7abf1e13/output/aarch64-linux-gnu/SHA256SUMS.part
1ececa8121d51a5c358e25e1a4b529413faadef5721387005db2b928c05cad3d  guix-build-a51d7abf1e13/output/aarch64-linux-gnu/bitcoin-a51d7abf1e13-aarch64-linux-gnu-debug.tar.gz
fbc1724995b2d9a43c0bccbb3c20e02d115d9aeaecce0792bde46d9fedc0dbe0  guix-build-a51d7abf1e13/output/aarch64-linux-gnu/bitcoin-a51d7abf1e13-aarch64-linux-gnu.tar.gz
8a9edb553d6e6db1d63e286bb972086f0a775f1b589eb81e108c4d382d0a33da  guix-build-a51d7abf1e13/output/arm-linux-gnueabihf/SHA256SUMS.part
b261a02f701635bd17abf75af99b1b9f8a474d3e2e5554218d3ed3e06e35c5cc  guix-build-a51d7abf1e13/output/arm-linux-gnueabihf/bitcoin-a51d7abf1e13-arm-linux-gnueabihf-debug.tar.gz
3594d2df18a28642e0e5a4459421179ce8b2826c62f3ad5e087e1fdc9911c9a4  guix-build-a51d7abf1e13/output/arm-linux-gnueabihf/bitcoin-a51d7abf1e13-arm-linux-gnueabihf.tar.gz
b6e9ac4d00fb837e314607cb2618c5ae9e85b209dd8dfa0a7a0be941a58bb8a4  guix-build-a51d7abf1e13/output/arm64-apple-darwin/SHA256SUMS.part
8281d236e2f33bb313441e888bfa0d84c8b0dfa636184c383b2958dc1e883f8c  guix-build-a51d7abf1e13/output/arm64-apple-darwin/bitcoin-a51d7abf1e13-arm64-apple-darwin-unsigned.dmg
d2ac774f35f2459a970b9d5536367d90e9c75b62800beb666529e92c417b47d0  guix-build-a51d7abf1e13/output/arm64-apple-darwin/bitcoin-a51d7abf1e13-arm64-apple-darwin-unsigned.tar.gz
11cb4660ba02cae997656cf0bf37cb47882c892dc4134d7d2e26d8f3c7648476  guix-build-a51d7abf1e13/output/arm64-apple-darwin/bitcoin-a51d7abf1e13-arm64-apple-darwin.tar.gz
7679e0643d5868e1fc4feed71f3d4b8e9178a7fc560a153fb4205f0586901c75  guix-build-a51d7abf1e13/output/dist-archive/bitcoin-a51d7abf1e13.tar.gz
ba860e8c30da71409d6281779cd7a2ad98afa2f9e395887d5635ca5a06e02697  guix-build-a51d7abf1e13/output/powerpc64-linux-gnu/SHA256SUMS.part
dc4d10e2bf00d2403bd2f90763d60e51eba4f4f754b934bdbdce8ac57291d0c1  guix-build-a51d7abf1e13/output/powerpc64-linux-gnu/bitcoin-a51d7abf1e13-powerpc64-linux-gnu-debug.tar.gz
aaf849c0db6e047f36bc98bbe4a8b9f9f2abb8144844160bba79c303b7532be4  guix-build-a51d7abf1e13/output/powerpc64-linux-gnu/bitcoin-a51d7abf1e13-powerpc64-linux-gnu.tar.gz
ddab779b72f8201f79945f68b9066fbb3f93d705efd7361ee6c71ed612933e5b  guix-build-a51d7abf1e13/output/powerpc64le-linux-gnu/SHA256SUMS.part
d8cbedb7da7f2b45e36da334421d380046fc1b2218417bed0966721d13cd6c7b  guix-build-a51d7abf1e13/output/powerpc64le-linux-gnu/bitcoin-a51d7abf1e13-powerpc64le-linux-gnu-debug.tar.gz
29b50d5e4f548b72a3df9e8a4e7d48b5eb5995526a94783d453caa6528eb8796  guix-build-a51d7abf1e13/output/powerpc64le-linux-gnu/bitcoin-a51d7abf1e13-powerpc64le-linux-gnu.tar.gz
ffd40756fdc58ee408adde791b0746b6388f8beef37a25c394d6a2c43534ce8c  guix-build-a51d7abf1e13/output/riscv64-linux-gnu/SHA256SUMS.part
86a61895adc86ee45f2e15ff5b1dfd86dba8a7cfefa0d35729c167a91d9735e4  guix-build-a51d7abf1e13/output/riscv64-linux-gnu/bitcoin-a51d7abf1e13-riscv64-linux-gnu-debug.tar.gz
a35b4619b435775bda3507af775a00f48cb54d02b2fdf227ae02e2b13b4b60b7  guix-build-a51d7abf1e13/output/riscv64-linux-gnu/bitcoin-a51d7abf1e13-riscv64-linux-gnu.tar.gz
05b7fedbb46b7199fdbd8726054acfc474574175b32fefd21f96169b070d03aa  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/SHA256SUMS.part
1a4215bd089041249f995de413a3627340f27dfa83477bc8a937954ce4bd0d29  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/bitcoin-a51d7abf1e13-x86_64-apple-darwin-unsigned.dmg
b6862e2cfb40eeb214852f9f6b27e59a0fcc12d5fc1e215bb20f595fa74f856c  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/bitcoin-a51d7abf1e13-x86_64-apple-darwin-unsigned.tar.gz
87396d48917a62f9851cc044bd6f0535f1ba60ee532feebfb79882ba03248050  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/bitcoin-a51d7abf1e13-x86_64-apple-darwin.tar.gz
9aa6ec4e7e28348cbb013582a278ee6888cba6aac8592299adddb15b5e50b9a7  guix-build-a51d7abf1e13/output/x86_64-linux-gnu/SHA256SUMS.part
d46f0c44863f838f079cf9f3d61877129074cda920a6b52f1e7bc24280ce52ba  guix-build-a51d7abf1e13/output/x86_64-linux-gnu/bitcoin-a51d7abf1e13-x86_64-linux-gnu-debug.tar.gz
adb5d445e390db0f738f92e8ac4b55814c816ce374975d129e281e5097c45d79  guix-build-a51d7abf1e13/output/x86_64-linux-gnu/bitcoin-a51d7abf1e13-x86_64-linux-gnu.tar.gz
d2ef49496929675e5ce2fa97490af52a0e205a71f80216fbf82224697abfc936  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/SHA256SUMS.part
2b3a37788e4d05fbf31ec41c94c68a8069cb4a90e7a4473b5cca970198b899c7  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64-debug.zip
03b2f27204c0d1616cba0addbda2bd3e820e34dc8a04ddaf244b024b4568acea  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64-setup-unsigned.exe
c8410e5ffcafd792c921c5f065c3bfc6340a7ed08df0d1423387edc264fa3b77  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64-unsigned.tar.gz
58fa2459e61b593cf8a545118406ca58e312722866cbbadab68a7d38883292de  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64.zip
```

hebasto commented:
Guix builds:
```
6edd213a90a043d09da144b99332ab16a320774098d35de28464262b25f260b2  guix-build-a51d7abf1e13/output/aarch64-linux-gnu/SHA256SUMS.part
1ececa8121d51a5c358e25e1a4b529413faadef5721387005db2b928c05cad3d  guix-build-a51d7abf1e13/output/aarch64-linux-gnu/bitcoin-a51d7abf1e13-aarch64-linux-gnu-debug.tar.gz
fbc1724995b2d9a43c0bccbb3c20e02d115d9aeaecce0792bde46d9fedc0dbe0  guix-build-a51d7abf1e13/output/aarch64-linux-gnu/bitcoin-a51d7abf1e13-aarch64-linux-gnu.tar.gz
8a9edb553d6e6db1d63e286bb972086f0a775f1b589eb81e108c4d382d0a33da  guix-build-a51d7abf1e13/output/arm-linux-gnueabihf/SHA256SUMS.part
b261a02f701635bd17abf75af99b1b9f8a474d3e2e5554218d3ed3e06e35c5cc  guix-build-a51d7abf1e13/output/arm-linux-gnueabihf/bitcoin-a51d7abf1e13-arm-linux-gnueabihf-debug.tar.gz
3594d2df18a28642e0e5a4459421179ce8b2826c62f3ad5e087e1fdc9911c9a4  guix-build-a51d7abf1e13/output/arm-linux-gnueabihf/bitcoin-a51d7abf1e13-arm-linux-gnueabihf.tar.gz
b6e9ac4d00fb837e314607cb2618c5ae9e85b209dd8dfa0a7a0be941a58bb8a4  guix-build-a51d7abf1e13/output/arm64-apple-darwin/SHA256SUMS.part
8281d236e2f33bb313441e888bfa0d84c8b0dfa636184c383b2958dc1e883f8c  guix-build-a51d7abf1e13/output/arm64-apple-darwin/bitcoin-a51d7abf1e13-arm64-apple-darwin-unsigned.dmg
d2ac774f35f2459a970b9d5536367d90e9c75b62800beb666529e92c417b47d0  guix-build-a51d7abf1e13/output/arm64-apple-darwin/bitcoin-a51d7abf1e13-arm64-apple-darwin-unsigned.tar.gz
11cb4660ba02cae997656cf0bf37cb47882c892dc4134d7d2e26d8f3c7648476  guix-build-a51d7abf1e13/output/arm64-apple-darwin/bitcoin-a51d7abf1e13-arm64-apple-darwin.tar.gz
7679e0643d5868e1fc4feed71f3d4b8e9178a7fc560a153fb4205f0586901c75  guix-build-a51d7abf1e13/output/dist-archive/bitcoin-a51d7abf1e13.tar.gz
ba860e8c30da71409d6281779cd7a2ad98afa2f9e395887d5635ca5a06e02697  guix-build-a51d7abf1e13/output/powerpc64-linux-gnu/SHA256SUMS.part
dc4d10e2bf00d2403bd2f90763d60e51eba4f4f754b934bdbdce8ac57291d0c1  guix-build-a51d7abf1e13/output/powerpc64-linux-gnu/bitcoin-a51d7abf1e13-powerpc64-linux-gnu-debug.tar.gz
aaf849c0db6e047f36bc98bbe4a8b9f9f2abb8144844160bba79c303b7532be4  guix-build-a51d7abf1e13/output/powerpc64-linux-gnu/bitcoin-a51d7abf1e13-powerpc64-linux-gnu.tar.gz
ddab779b72f8201f79945f68b9066fbb3f93d705efd7361ee6c71ed612933e5b  guix-build-a51d7abf1e13/output/powerpc64le-linux-gnu/SHA256SUMS.part
d8cbedb7da7f2b45e36da334421d380046fc1b2218417bed0966721d13cd6c7b  guix-build-a51d7abf1e13/output/powerpc64le-linux-gnu/bitcoin-a51d7abf1e13-powerpc64le-linux-gnu-debug.tar.gz
29b50d5e4f548b72a3df9e8a4e7d48b5eb5995526a94783d453caa6528eb8796  guix-build-a51d7abf1e13/output/powerpc64le-linux-gnu/bitcoin-a51d7abf1e13-powerpc64le-linux-gnu.tar.gz
ffd40756fdc58ee408adde791b0746b6388f8beef37a25c394d6a2c43534ce8c  guix-build-a51d7abf1e13/output/riscv64-linux-gnu/SHA256SUMS.part
86a61895adc86ee45f2e15ff5b1dfd86dba8a7cfefa0d35729c167a91d9735e4  guix-build-a51d7abf1e13/output/riscv64-linux-gnu/bitcoin-a51d7abf1e13-riscv64-linux-gnu-debug.tar.gz
a35b4619b435775bda3507af775a00f48cb54d02b2fdf227ae02e2b13b4b60b7  guix-build-a51d7abf1e13/output/riscv64-linux-gnu/bitcoin-a51d7abf1e13-riscv64-linux-gnu.tar.gz
05b7fedbb46b7199fdbd8726054acfc474574175b32fefd21f96169b070d03aa  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/SHA256SUMS.part
1a4215bd089041249f995de413a3627340f27dfa83477bc8a937954ce4bd0d29  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/bitcoin-a51d7abf1e13-x86_64-apple-darwin-unsigned.dmg
b6862e2cfb40eeb214852f9f6b27e59a0fcc12d5fc1e215bb20f595fa74f856c  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/bitcoin-a51d7abf1e13-x86_64-apple-darwin-unsigned.tar.gz
87396d48917a62f9851cc044bd6f0535f1ba60ee532feebfb79882ba03248050  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/bitcoin-a51d7abf1e13-x86_64-apple-darwin.tar.gz
9aa6ec4e7e28348cbb013582a278ee6888cba6aac8592299adddb15b5e50b9a7  guix-build-a51d7abf1e13/output/x86_64-linux-gnu/SHA256SUMS.part
d46f0c44863f838f079cf9f3d61877129074cda920a6b52f1e7bc24280ce52ba  guix-build-a51d7abf1e13/output/x86_64-linux-gnu/bitcoin-a51d7abf1e13-x86_64-linux-gnu-debug.tar.gz
adb5d445e390db0f738f92e8ac4b55814c816ce374975d129e281e5097c45d79  guix-build-a51d7abf1e13/output/x86_64-linux-gnu/bitcoin-a51d7abf1e13-x86_64-linux-gnu.tar.gz
d2ef49496929675e5ce2fa97490af52a0e205a71f80216fbf82224697abfc936  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/SHA256SUMS.part
2b3a37788e4d05fbf31ec41c94c68a8069cb4a90e7a4473b5cca970198b899c7  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64-debug.zip
03b2f27204c0d1616cba0addbda2bd3e820e34dc8a04ddaf244b024b4568acea  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64-setup-unsigned.exe
c8410e5ffcafd792c921c5f065c3bfc6340a7ed08df0d1423387edc264fa3b77  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64-unsigned.tar.gz
58fa2459e61b593cf8a545118406ca58e312722866cbbadab68a7d38883292de  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64.zip
```

fanquake commented:
Guix Build:
```bash
6edd213a90a043d09da144b99332ab16a320774098d35de28464262b25f260b2  guix-build-a51d7abf1e13/output/aarch64-linux-gnu/SHA256SUMS.part
1ececa8121d51a5c358e25e1a4b529413faadef5721387005db2b928c05cad3d  guix-build-a51d7abf1e13/output/aarch64-linux-gnu/bitcoin-a51d7abf1e13-aarch64-linux-gnu-debug.tar.gz
fbc1724995b2d9a43c0bccbb3c20e02d115d9aeaecce0792bde46d9fedc0dbe0  guix-build-a51d7abf1e13/output/aarch64-linux-gnu/bitcoin-a51d7abf1e13-aarch64-linux-gnu.tar.gz
8a9edb553d6e6db1d63e286bb972086f0a775f1b589eb81e108c4d382d0a33da  guix-build-a51d7abf1e13/output/arm-linux-gnueabihf/SHA256SUMS.part
b261a02f701635bd17abf75af99b1b9f8a474d3e2e5554218d3ed3e06e35c5cc  guix-build-a51d7abf1e13/output/arm-linux-gnueabihf/bitcoin-a51d7abf1e13-arm-linux-gnueabihf-debug.tar.gz
3594d2df18a28642e0e5a4459421179ce8b2826c62f3ad5e087e1fdc9911c9a4  guix-build-a51d7abf1e13/output/arm-linux-gnueabihf/bitcoin-a51d7abf1e13-arm-linux-gnueabihf.tar.gz
b6e9ac4d00fb837e314607cb2618c5ae9e85b209dd8dfa0a7a0be941a58bb8a4  guix-build-a51d7abf1e13/output/arm64-apple-darwin/SHA256SUMS.part
8281d236e2f33bb313441e888bfa0d84c8b0dfa636184c383b2958dc1e883f8c  guix-build-a51d7abf1e13/output/arm64-apple-darwin/bitcoin-a51d7abf1e13-arm64-apple-darwin-unsigned.dmg
d2ac774f35f2459a970b9d5536367d90e9c75b62800beb666529e92c417b47d0  guix-build-a51d7abf1e13/output/arm64-apple-darwin/bitcoin-a51d7abf1e13-arm64-apple-darwin-unsigned.tar.gz
11cb4660ba02cae997656cf0bf37cb47882c892dc4134d7d2e26d8f3c7648476  guix-build-a51d7abf1e13/output/arm64-apple-darwin/bitcoin-a51d7abf1e13-arm64-apple-darwin.tar.gz
7679e0643d5868e1fc4feed71f3d4b8e9178a7fc560a153fb4205f0586901c75  guix-build-a51d7abf1e13/output/dist-archive/bitcoin-a51d7abf1e13.tar.gz
ba860e8c30da71409d6281779cd7a2ad98afa2f9e395887d5635ca5a06e02697  guix-build-a51d7abf1e13/output/powerpc64-linux-gnu/SHA256SUMS.part
dc4d10e2bf00d2403bd2f90763d60e51eba4f4f754b934bdbdce8ac57291d0c1  guix-build-a51d7abf1e13/output/powerpc64-linux-gnu/bitcoin-a51d7abf1e13-powerpc64-linux-gnu-debug.tar.gz
aaf849c0db6e047f36bc98bbe4a8b9f9f2abb8144844160bba79c303b7532be4  guix-build-a51d7abf1e13/output/powerpc64-linux-gnu/bitcoin-a51d7abf1e13-powerpc64-linux-gnu.tar.gz
ddab779b72f8201f79945f68b9066fbb3f93d705efd7361ee6c71ed612933e5b  guix-build-a51d7abf1e13/output/powerpc64le-linux-gnu/SHA256SUMS.part
d8cbedb7da7f2b45e36da334421d380046fc1b2218417bed0966721d13cd6c7b  guix-build-a51d7abf1e13/output/powerpc64le-linux-gnu/bitcoin-a51d7abf1e13-powerpc64le-linux-gnu-debug.tar.gz
29b50d5e4f548b72a3df9e8a4e7d48b5eb5995526a94783d453caa6528eb8796  guix-build-a51d7abf1e13/output/powerpc64le-linux-gnu/bitcoin-a51d7abf1e13-powerpc64le-linux-gnu.tar.gz
ffd40756fdc58ee408adde791b0746b6388f8beef37a25c394d6a2c43534ce8c  guix-build-a51d7abf1e13/output/riscv64-linux-gnu/SHA256SUMS.part
86a61895adc86ee45f2e15ff5b1dfd86dba8a7cfefa0d35729c167a91d9735e4  guix-build-a51d7abf1e13/output/riscv64-linux-gnu/bitcoin-a51d7abf1e13-riscv64-linux-gnu-debug.tar.gz
a35b4619b435775bda3507af775a00f48cb54d02b2fdf227ae02e2b13b4b60b7  guix-build-a51d7abf1e13/output/riscv64-linux-gnu/bitcoin-a51d7abf1e13-riscv64-linux-gnu.tar.gz
05b7fedbb46b7199fdbd8726054acfc474574175b32fefd21f96169b070d03aa  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/SHA256SUMS.part
1a4215bd089041249f995de413a3627340f27dfa83477bc8a937954ce4bd0d29  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/bitcoin-a51d7abf1e13-x86_64-apple-darwin-unsigned.dmg
b6862e2cfb40eeb214852f9f6b27e59a0fcc12d5fc1e215bb20f595fa74f856c  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/bitcoin-a51d7abf1e13-x86_64-apple-darwin-unsigned.tar.gz
87396d48917a62f9851cc044bd6f0535f1ba60ee532feebfb79882ba03248050  guix-build-a51d7abf1e13/output/x86_64-apple-darwin/bitcoin-a51d7abf1e13-x86_64-apple-darwin.tar.gz
9aa6ec4e7e28348cbb013582a278ee6888cba6aac8592299adddb15b5e50b9a7  guix-build-a51d7abf1e13/output/x86_64-linux-gnu/SHA256SUMS.part
d46f0c44863f838f079cf9f3d61877129074cda920a6b52f1e7bc24280ce52ba  guix-build-a51d7abf1e13/output/x86_64-linux-gnu/bitcoin-a51d7abf1e13-x86_64-linux-gnu-debug.tar.gz
adb5d445e390db0f738f92e8ac4b55814c816ce374975d129e281e5097c45d79  guix-build-a51d7abf1e13/output/x86_64-linux-gnu/bitcoin-a51d7abf1e13-x86_64-linux-gnu.tar.gz
d2ef49496929675e5ce2fa97490af52a0e205a71f80216fbf82224697abfc936  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/SHA256SUMS.part
2b3a37788e4d05fbf31ec41c94c68a8069cb4a90e7a4473b5cca970198b899c7  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64-debug.zip
03b2f27204c0d1616cba0addbda2bd3e820e34dc8a04ddaf244b024b4568acea  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64-setup-unsigned.exe
c8410e5ffcafd792c921c5f065c3bfc6340a7ed08df0d1423387edc264fa3b77  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64-unsigned.tar.gz
58fa2459e61b593cf8a545118406ca58e312722866cbbadab68a7d38883292de  guix-build-a51d7abf1e13/output/x86_64-w64-mingw32/bitcoin-a51d7abf1e13-win64.zip
```

Sjors commented:
I get the same hashes.
